src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2781,7 +2781,26 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
             $config['dsn'] = 'smtp://null';
         }
         $transports = $config['dsn'] ? ['main' => $config['dsn']] : $config['transports'];
-        $container->getDefinition('mailer.transports')->setArgument(0, $transports);
+        $container->getDefinition('mailer.transports')->setArgument(0, array_combine(array_keys($config['transports']), array_column($config['transports'], 'dsn')));
+
+        $transportRateLimiterReferences = [];
+
+        foreach ($transports as $name => $transport) {
+            if ($transport['rate_limiter']) {
+                if (!interface_exists(LimiterInterface::class)) {
```

Comment:
This `if` could be moved outside the `for` loop.
The verification can be done just after if `$transportRateLimiterReferences` is not empty and the interface is missing.

Reply:
Yes - however, the same would apply to this existing code: 

https://github.com/symfony/symfony/blob/3ec30b693fee9a37cdae6718259395eba2e8b25e/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L2458-L2460

Do you want me to change it in the unrelated section as well?